include/ignition/math/Interval.hh

```diff
@@ -0,0 +1,292 @@
+      /// \return true if it is contained, false otherwise
+      public: bool Contains(const T &_value) const
+      {
+        if (this->leftClosed && this->rightClosed)
+        {
+          return this->leftValue <= _value && _value <= this->rightValue;
+        }
+        if (this->leftClosed)
+        {
+          return this->leftValue <= _value && _value < this->rightValue;
+        }
+        if (this->rightClosed)
+        {
+          return this->leftValue < _value && _value <= this->rightValue;
+        }
+        return this->leftValue < _value && _value < this->rightValue;
+      }
+
+      /// \brief Check if the interval contains `_other` interval
+      /// \param[in] _other interval to check for membership
+      /// \return true if it is contained, false otherwise
+      public: bool Contains(const Interval<T> &_other) const
+      {
+        if (this->Empty() || _other.Empty())
+        {
+          return false;
+        }
+        if (!this->leftClosed && _other.leftClosed)
+        {
+          if (_other.leftValue <= this->leftValue)
+          {
+            return false;
+          }
+        }
+        else
+        {
+          if (_other.leftValue < this->leftValue)
+          {
+            return false;
+          }
+        }
+        if (!this->rightClosed && _other.rightClosed)
+        {
+          if (this->rightValue <= _other.rightValue)
+          {
+            return false;
+          }
+        }
+        else
+        {
+          if (this->rightValue < _other.rightValue)
+          {
+            return false;
+          }
+        }
+        return true;
+      }
+
+      /// \brief Check if the interval intersects `_other` interval
+      /// \param[in] _other interval to check for intersection
+      /// \return true if both intervals intersect, false otherwise
+      public: bool Intersects(const Interval<T> &_other) const
+      {
+        if (this->Empty() || _other.Empty())
+        {
+          return false;
+        }
+        if (this->rightClosed && _other.leftClosed)
+        {
+          if (this->rightValue < _other.leftValue)
+          {
+            return false;
+          }
+        }
+        else
+        {
+          if (this->rightValue <= _other.leftValue)
+          {
+            return false;
+          }
+        }
+        if (_other.rightClosed && this->leftClosed)
+        {
+          if (_other.rightValue < this->leftValue)
+          {
+            return false;
+          }
+        }
+        else
+        {
+          if (_other.rightValue <= this->leftValue)
+          {
+            return false;
+          }
+        }
+        return true;
+      }
+
+      /// \brief Equality test operator
+      /// \param _other interval to check for equality
+      /// \return true if intervals are equal, false otherwise
+      public: bool operator==(const Interval<T> &_other) const
+      {
+        return this->Contains(_other) && _other.Contains(*this);
```

Comment:
Just curious, this results in more comparisons than directly comparing the 4 member variables. Does this cover a case that wouldn't be covered otherwise, or is it just a more formal way of comparing intervals?

Reply:
@chapulina this allows me to define equality without specifying an arbitrary tolerance. Checking floating-point numbers for equality has a few rough edges (and any attempt to do so generates a warning when building Ignition Math).

Reply:
Makes sense, thanks!